packages/core/src/Stores/contract-store.js

```diff
@@ -299,7 +322,33 @@ export default class ContractStore extends BaseStore {
                 );
 
                 main_barrier.updateBarrierShade(true, contract_type);
+
+                barriers = [main_barrier];
+            } else if (isResetContract(contract_type) && entry_spot) {
+                const main_barrier = new ChartBarrierStore(entry_spot, low_barrier, null, {
+                    color: is_dark_mode ? BARRIER_COLORS.DARK_GRAY : BARRIER_COLORS.GRAY,
+                    line_style: BARRIER_LINE_STYLES.SOLID,
+                    not_draggable: true,
+                    hideBarrierLine: false,
+                    shade: DEFAULT_SHADES['2'],
+                });
+
+                main_barrier.updateBarrierShade(true, contract_type);
+
                 barriers = [main_barrier];
+
+                if (reset_time) {
+                    const reset_barrier_object = new ChartBarrierStore(reset_barrier, low_barrier, null, {
```

Comment:
```suggestion
                    const reset_barrier = new ChartBarrierStore(reset_barrier, low_barrier, null, {
```

Reply:
We already have `reset_barrier` in this line: `new ChartBarrierStore(reset_barrier`,  ..